File: app/src/main/java/com/hhh/paws/database/viewModel/IdentificationViewModel.java
package com.hhh.paws.database.viewModel;

import androidx.lifecycle.ViewModel;

import com.hhh.paws.database.model.Identification;
import com.hhh.paws.database.repository.IdentificationRepository;

import javax.inject.Inject;

import dagger.hilt.android.lifecycle.HiltViewModel;
import io.reactivex.rxjava3.core.Observable;


@HiltViewModel
public class IdentificationViewModel extends ViewModel {

    private IdentificationRepository repository;
    @Inject
    public IdentificationViewModel(IdentificationRepository repository) {
        this.repository = repository;
    }

    public Observable<Identification> getIdentification(String petName) {
        return repository.getIdentification(petName);
    }

    public void setIdentification(String petName, Identification identification) {
        repository.setIdentification(petName, identification);
    }
}

File: app/src/main/java/com/hhh/paws/database/viewModel/IdentificationViewModel.kt
package com.hhh.paws.database.viewModel

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.hhh.paws.database.model.Identification
import com.hhh.paws.database.repository.IdentificationRepository
import com.hhh.paws.util.UiState
import dagger.hilt.android.lifecycle.HiltViewModel
import kotlinx.coroutines.launch
import javax.inject.Inject


@HiltViewModel
class IdentificationViewModel @Inject constructor(private val repository: IdentificationRepository): ViewModel() {

    private var _identificationInstance = MutableLiveData<UiState<Identification>>()
    val identificationInstance: LiveData<UiState<Identification>> get() = _identificationInstance
    fun getIdentificationInstance(petName: String) {
        _identificationInstance.value = UiState.Loading
        viewModelScope.launch {
            repository.getIdentification(petName) {
                _identificationInstance.value = it
            }
        }
    }

    private var _identificationAdd = MutableLiveData<UiState<String>>()
    val identificationAdd: LiveData<UiState<String>> get() = _identificationAdd
    fun setIdentificationInstance(petName: String, identification: Identification) {
        _identificationAdd.value = UiState.Loading
        viewModelScope.launch {
            repository.setIdentification(petName, identification) {
                _identificationAdd.value = it
            }
        }
    }
}